cli/server_slim.go

```diff
@@ -0,0 +1,65 @@
+//go:build slim
+
+package cli
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/coder/coder/cli/deployment"
+	"github.com/coder/coder/coderd"
```

Comment:
Importing `coderd` here is unfortunate. I think we could reduce binary size if we didn't but I might be wrong. I think we can avoid it if we change enterprise slim too?

Reply:
Sadly also `root.go` uses this function so the signature needs to match on both. I think this could be done as a second improvement, but I want to get this in so we can unblock the community PR #5738

Reply:
Ah, didn't think of that. Yes makes sense 👍🏻.